packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -582,12 +582,17 @@ export async function createPluginContainer(
           'handler' in plugin.resolveId
             ? plugin.resolveId.handler
             : plugin.resolveId
-        const result = await handler.call(ctx as any, rawId, importer, {
-          custom: options?.custom,
-          isEntry: !!options?.isEntry,
-          ssr,
-          scan
-        })
+        const result = await handler.call(
+          ctx as any,
+          rawId.replace(/%20/g, ' '),
```

Comment:
can we just use `decodeURI(rawId)`

Reply:
https://github.com/vitest-dev/vitest/pull/1952/files#diff-0d08494797e08cc20a3fd0f24593279cb51d179486a6188d86b92d4855a23633R224

Reply:
failed to pass the test by using decodeURI[#8498](https://github.com/vitejs/vite/issues/8498)